drivers/hmis_csv_importer/app/models/hmis_csv_importer/hmis_csv_cleanup/append_project_id.rb

```diff
@@ -0,0 +1,50 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module HmisCsvImporter::HmisCsvCleanup
+  class AppendProjectId < Base
+    def cleanup!
+      project_batch = []
+
+      project_scope.find_each do |project|
+        project.ProjectName = "#{project.ProjectName} (#{project.ProjectID})"
+        project.set_source_hash
+        project_batch << project
+      end
+
+      project_source.import(
+        project_batch,
+        on_duplicate_key_update: {
+          conflict_target: conflict_target(project_source),
+          columns: [:ProjectName, :source_hash],
+        },
+      )
```

Comment:
This is probably fine, for very large number of projects we'd probably want to batch upserts too

```suggestion
      project_scope.find_in_batches do |project_batch|
        project_batch.each do |project|
          project.ProjectName = "#{project.ProjectName} (#{project.ProjectID})"
          project.set_source_hash
        end
    
        project_source.import(
          project_batch,
          on_duplicate_key_update: {
            conflict_target: conflict_target(project_source),
            columns: [:ProjectName, :source_hash],
          },
        )
      end
```